### PR TITLE
fix(api): separate disengage and stop

### PR DESCRIPTION
**Pull request by ahiuchingau**
# Overview
When we issue a StopRequest to the Flex axes, the following happens on the firmware side:
1. the motor gets disengaged to stop any movement
2. any queued up motor tasks are cleared

This has worked pretty well for immediately halting robot motion. But, this also means sometimes we're unexpectedly disengaging the motors when not necessary and leaving the axes in a non-movable state. 

We can solve this by not disengaging the motors in the StopRequest task and let the DisableMotorRequest handles that. You'll need the changes in https://github.com/Opentrons/ot3-firmware/pull/710 in order for this to work properly.

### CODE REVIEW

File: api/src/opentrons/hardware_control/ot3api.py [hunk 1]
```diff
@@ -692,14 +692,25 @@ async def _chained_calls() -> None:
 
         asyncio.run_coroutine_threadsafe(_chained_calls(), self._loop)
 
-    async def halt(self) -> None:
-        """Immediately stop motion."""
+    async def stop_motors(self) -> None:
+        """Immediately stop motors."""
         await self._backend.halt()
+
+    def stop_modules(self) -> None:
```
Comment (Laura-Danielle) [inline api/src/opentrons/hardware_control/ot3api.py:699]: What is this guy meant to do?

File: api/src/opentrons/hardware_control/ot3api.py [hunk 2]
```diff
@@ -692,14 +692,25 @@ async def _chained_calls() -> None:
...
+        """Immediately stop modules."""
         asyncio.run_coroutine_threadsafe(self._execution_manager.cancel(), self._loop)
 
+    async def halt(self) -> None:
+        """Immediately disengage all present motors and clear motor and module tasks."""
+        await self.disengage_axes(
+            [ax for ax in Axis if self._backend.axis_is_present(ax)]
+        )
+        await self.stop_motors()
```
Comment (Laura-Danielle) [inline api/src/opentrons/hardware_control/ot3api.py:708]: Should the order be
```
await self.stop_motors()
await self.disengage_axes()
```
?

**Review by Laura-Danielle — COMMENTED** [review]
Leaving as a comment. Don't want my q's to be blocking.

File: api/src/opentrons/hardware_control/ot3api.py (hunk 2 above)
Comment (ahiuchingau) [reply api/src/opentrons/hardware_control/ot3api.py:708]: The disengage request is used to immediately stop the motor while it's still in motion. This allows the halt request to behave similarly to an estop.

File: api/src/opentrons/hardware_control/ot3api.py (hunk 1 above)
Comment (ahiuchingau) [reply api/src/opentrons/hardware_control/ot3api.py:699]: This clears any future module tasks scheduled by the protocol

Comment (Laura-Danielle) [reply api/src/opentrons/hardware_control/ot3api.py:699]: Ah yes I see it now. The diff on github was being weird for me and I thought it was a free floating func

**Review by Laura-Danielle — APPROVED** [review]
lgtm

**Pull request merged**